app/server.py

```diff
@@ -1,40 +1,179 @@
 
-bot = Bot(token=API_TOKEN)
-dp = Dispatcher(bot)
+def debug_requests(f):
+    """
+    Telegram event decorator
+    """
+    def inner(*args, **kwargs):
+        try:
+            logger.info(f"Function call {f.__name__}")
+            return f(*args, **kwargs)
+        except Exception:
+            logger.exception(f"Error in handler {f.__name__}")
+            raise
+    return inner
+
+API_TOKEN = os.getenv("TELEGRAM_API_TOKEN")
+
+"""
+Handler for chat buttons
+"""
+def keyboard_callback_handler(update, context):
+    query = update.callback_query
+    query.answer()
+    data = query.data
+    now = datetime.datetime.now()
+    chat_id = update.effective_message.chat_id
+    current_text =update.effective_message.text
+    if data == CALLBACK_BUTTON1:
+        context.bot.send_message(
+            chat_id=chat_id,
+            text="Add a new contact to send a request or \n"
+                 "choose from the list with whom you have already worked",
+            reply_markup=get_add_debt_keyboard(),
+        )
+    elif data == CALLBACK_BUTTON2:
+        context.bot.send_message(
+            chat_id=chat_id,
+            text=" There will be something like a menu in which it will be visible to whom you have already sent requests \n"
+                 "Each button below the message will connect you with the contact you have already contacted.\n"
+                 "I also don’t know how to implementь",
+            reply_markup=get_debt_keyboard_inline(),
+        )
+    elif data == CALLBACK_BUTTON5_BACK:
+        context.bot.send_message(
+            chat_id=chat_id,
+            text="Choose an action",
+            reply_markup=get_keyboard_inline()
+        )
 
-dp.middleware.setup(AccessMiddleware(ADMIN_ID))
+@debug_requests
+def add_debt(update, context):
+    context.bot.send_message(
+        chat_id=update.message.chat_id,
+        text="Add a new contact or select an existing one",
+        reply_markup=get_keyboard_inline()
+    )
 
+@debug_requests
+def pay_debt(update, context):
+    context.bot.send_message(
+        chat_id=update.message.chat_id,
+        text="A list of lenders for payment will be displayed.",
+        reply_markup=get_debt_keyboard_inline()
+    )
 
-@dp.message_handler(commands=['start', 'help'])
-async def send_welcome(message: types.Message):
-    await message.answer("Hi! I'm BillOfExchangeBot!\n\
-    /help - help\n\
-    /my_id - get my telegram id\n")
+@debug_requests
+def do_start(update, context):
+    #chat_id =475038836
+    text = "Hello! This is a bot for tracking, sharing your purchases / accounts \n" \
+           "For information, click the Information button."
+    context.bot.send_message(
+        chat_id=update.message.chat_id,
+        text=text,
+        reply_markup=get_start_keyboard()
+    )
 
+@debug_requests
+def do_echo(update, context):
+    username = context.user_data
+    chat_id = update.message.chat_id
+    text = update.message.text
+    if text == BUTTON1_ADD_DEBT:
```

Comment:
It's better to use switch-case construction with alike cases.

Reply:
fixed it